### PR TITLE
fix: change date picker to compose one

### PR DESCRIPTION
**Pull request by rBergCode**
# Summary
The issue was that the calendar had ugly colors. We needed to use the primary color.

# Details
As I investigated the issue, I found out we used the wrong `DatePickerDialog`. I changed it to use the compose dialog one.
Also, I updated the them in order to use the primary color app-wide. This changes most of the colors to use the correct one.

### CODE REVIEW

File: app/src/main/java/ch/hikemate/app/ui/map/HikeDetailScreen.kt [hunk 1]
```diff
@@ -426,7 +443,7 @@ fun DateDetailRow(
                   // saved Date
                   style = MaterialTheme.typography.bodySmall,
                   modifier =
-                      Modifier.clickable { showDatePickerDialog() }
+                      Modifier.clickable { showDatePicker() }
```
Comment (davidholzwarth) [inline app/src/main/java/ch/hikemate/app/ui/map/HikeDetailScreen.kt:446]: Request: LGMT, but I noticed that the border of the Box in which this date is is black. Could you change it to onPrimary please or something similar?

File: app/src/main/res/values/strings.xml [hunk 2]
```diff
@@ -63,6 +63,10 @@
     <string name="hike_detail_screen_value_planned">Planned</string>
     <string name="hike_detail_screen_add_a_date_button_text">Add a date</string>
 
+    <!-- Hike Details Screen - Date Picker -->
+    <string name="hike_detail_screen_date_picker_cancel_button">Cancel</string>
+    <string name="hike_detail_screen_date_picker_confirm_button">Plan my hike!</string>
```
Comment (davidholzwarth) [inline app/src/main/res/values/strings.xml:68]: Nitpick: Maybe `Plan this hike!` sounds better? Or just `Plan this hike`?

**Review by davidholzwarth — COMMENTED** [review]
Nice, looks way better. Two little nitpicks but otherwise you're good. :)

File: app/src/main/res/values/strings.xml (hunk 2 above)
Comment (rBergCode) [reply app/src/main/res/values/strings.xml:68]: "Plan this hike!" sounds good.

File: app/src/main/java/ch/hikemate/app/ui/map/HikeDetailScreen.kt (hunk 1 above)
Comment (davidholzwarth) [reply app/src/main/java/ch/hikemate/app/ui/map/HikeDetailScreen.kt:458]: Thanks

**Review by davidholzwarth — APPROVED** [review]
LGTM now!

**Pull request merged**